### PR TITLE
Add functions to compute disclosure for different stream types

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  def forFlatTransaction[Nid, Cid, Val](tx: GenTransaction[Nid, Cid, Val]): Relation[Nid, Party] =
+    tx.nodes.collect {
+      case (nodeId, c: NodeInfo.Create) =>
+        nodeId -> c.stakeholders
+      case (nodeId, e: NodeInfo.Exercise) if e.consuming =>
+        nodeId -> e.stakeholders
+    }
+
+  def forTransactionTree[Nid, Cid, Val](
+      disclosure: GenTransaction[Nid, Cid, Val] => Relation[Nid, Party])(
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: Why not pass in the disclosure relation directly?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    val createAndExercise: Set[Nid] =
+      tx.nodes.collect {
+        case p @ (_, _: NodeInfo.Create) => p
+        case p @ (_, _: NodeInfo.Exercise) => p
+      }.keySet
+
+    disclosure(tx).filterKeys(createAndExercise)
+  }
+
+  val forTransactionTree: Transaction => Relation[NodeId, Party] =
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: I'm curious why this is not a regular `def` that takes a parameter?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: I can turn this into one.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: I was hoping I could somehow make this testable by abstracting over this, but I guess I can undo this.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: Plus, initially I used this to test this in the current `TransactionConversion`.

Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: If yo don't intend to use it, then let's keep it to the minimum that's needed. 👍

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: Addressed by https://github.com/digital-asset/daml/pull/4970/commits/f8e11794c14a26621f54a77274d02ab8afe56067

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala]: Addressed by https://github.com/digital-asset/daml/pull/4970/commits/f8e11794c14a26621f54a77274d02ab8afe56067

**Pull request merged**